src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py

```diff
@@ -442,7 +443,15 @@ def _evaluate(  # pylint: disable=too-many-locals
     # Batch Run
     evaluators_info = {}
     use_pf_client = kwargs.get("_use_pf_client", True)
-    batch_run_client = ProxyClient(pf_client) if use_pf_client else CodeClient()
+    if use_pf_client:
+        batch_run_client = ProxyClient(pf_client)
+
+        # Ensure the absolute path is passed to pf.run, as relative path doesn't work with
+        # multiple evaluators. If the path is already absolute, abspath will return the original path.
+        data = os.path.abspath(data)

```

Comment:
Would this affect the data value that customer passed in ? If yes this could have unknown side affects in user code that data variable is being used somewhere else too

Reply:
Welcome back Ankit :) 
No, it won't. Here is the prove:

```
data = "hello"
def test(data):
    data = "world"

test(data)
print(data)
# "hello"
```